# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/ion/bug1877709.js
# Commit: e191d0048b49
# Full Hash: e191d0048b4905326fbc8002846517709dfe19be
# Author: Nicolas B. Pierron <nicolas.b.pierron@mozilla.com>
# Date: 2024-02-23 03:40:30
# Description:
#   Bug 1877357 - Insert coercion instruction in the precedessor. r=iain
#   
#   Previously, when resolving Phi's type, we would insert the coercion instruction
#   at the end of the block in which the instruction is.
#   
# ==============================================================================

diff -r 61af32f40777 -r e191d0048b49 js/src/jit-test/tests/ion/bug1877709.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/ion/bug1877709.js	Thu Feb 22 10:28:14 2024 +0000
@@ -0,0 +1,22 @@
+function testMathyFunction (f, inputs) {
+  var results = [];
+    for (var j = 0; j < inputs.length; ++j)
+      for (var k = 0; k < inputs.length; ++k)
+        results.push(f(inputs[j], inputs[k]));
+}
+mathy1=(function(stdlib,foreign,heap){
+  ff = foreign.ff;
+  Float32ArrayView = new stdlib.Float32Array(heap);
+  Uint32ArrayView = new stdlib.Uint32Array(heap);
+  function f(d0) {
+    var i2=0;
+    var i4;
+    i2=Float32ArrayView[2];
+    i4=i2;
+    ff(2,0) ? f : 6;
+    Uint32ArrayView[!!d0] + [...[eval]]
+    return i4 ? 1 : 0;
+  }
+return f
+})(this,{ ff:(Function('')) },new SharedArrayBuffer(40));
+testMathyFunction(mathy1,[Math.N,Number.M,(2),Number.M])